### DISCUSSION TITLE
Strategic use of Auro Library

### DISCUSSION THREAD

**Discussion by jordanjones243** [AlaskaAirlines/WC-Generator#445] [Announcements]
### General Support Request

We want to move most (if not all) templates that are currently living in the generator over into `auro-library`. We want to do this because we want users to be able to use the generator and generate a component, without having all of the Auro constraints on the newly generated component. A user then would be able to run a script from `auro-library` to add all of those Auro templates if they wanted them.

### Possible Solution

* Move all templates currently living in the generator to `auro-library`
* Sift through all of the templates and see what is necessary to the generator and what can be moved to `auro-library`, resulting in only some templates transferred

### Additional context

As part of moving the template files out of the generator the template *must* be updated to have a dev dependency of auro-library and incorporate a script into the package.json to run the scripting that incorporates all the Auro configuration files we have automated.

This will be a necessary step of creating a new auro component and the template should come out of the box ready to pull the configuration in via automation.

**Comment by blackfalcon** [comment]
This needs to be a discussion about the state of Auro Library and how we intend to use it as a tool. The current repo is kind of a mess and we need to address this from a content sanity perspective and an architectural perspective.
